roles/custom/matrix-bridge-hookshot/templates/systemd/matrix-hookshot.service.j2

```diff
@@ -30,6 +30,12 @@ ExecStart={{ devture_systemd_docker_base_host_command_docker }} run --rm --name
           {% endfor %}
           {{ matrix_hookshot_docker_image }}
 
+{% if matrix_hookshot_experimental_encryption_enabled %}
+ExecStartPre={{ devture_systemd_docker_base_host_command_docker }} network connect matrix-redis {{ matrix_hookshot_container_url }}
+{% endif %}
```

Comment:
As discussed before, it's not great for roles to make assumptions about other roles.

The assumption that this role makes there is that when encryption is enabled, Redis will also be enabled via nothing else but `matrix-redis`.

-----

It's better to redo this by:

- adding a new  `matrix_hookshot_container_additional_networks` variable, which can connect the Hookshot container to any arbitrary additional network. In some roles, we even split this into multiple lists of networks (multiple variables) in `defaults/main.yml`:
  - `matrix_hookshot_container_additional_networks: "{{ matrix_hookshot_container_additional_networks_auto + matrix_hookshot_container_additional_networks_custom }}"`
  - `matrix_hookshot_container_additional_networks_auto: []`
  - `matrix_hookshot_container_additional_networks_custom: []`
  - With the above, `matrix_hookshot_container_additional_networks_auto` is supposed to be managed by the playbook, while `matrix_hookshot_container_additional_networks_custom` is supposed to be managed by the user
- `matrix_hookshot_container_additional_networks_auto` may be redefined in `group_vars/matrix_servers` to inject `redis_container_network` (equal to `matrix-redis`) into the list of networks under certain conditions (if `matrix_hookshot_queue_host == redis_identifier and redis_enabled`). You can find various examples of `_additional_networks` redefinitions in `group_vars/matrix_servers`
- redoing this systemd `.service` file, so that it loops over the list of networks (`matrix_hookshot_container_additional_networks`) and connects to them all

With this, the role is configurable and does not hardcode anything about `matrix-redis`. All wiring happens in `group_vars/matrix_servers` and people can do their own network wiring or use external Redis if they need to.